### ISSUE TITLE
create `src/errors.rs` with generic errors for the entire crate

### ISSUE THREAD

**Issue by dlaehnemann** [rust-bio/rust-htslib#215]
In PR #165 (now superseded by PR #214 ), the review led to the idea of defining generic errors like `FileNotFound` and `NonUnicodePath` on the highest level of the crate, i.e. in a new file `src/errors.rs`. This avoids redundantly defining such errors for all submodules. Currently, the two example errors are defined both for `bam`, `tbx` and `faidx`:

https://github.com/rust-bio/rust-htslib/blob/61b0644b73ab40612cd23e104ff0a3fe647bf064/src/bam/errors.rs#L21-L24

https://github.com/rust-bio/rust-htslib/blob/61b0644b73ab40612cd23e104ff0a3fe647bf064/src/tbx/errors.rs#L15-L18

https://github.com/rust-bio/rust-htslib/blob/29565f9edda5ee71e7ee6547128a76ff29b2f404/src/faidx/errors.rs#L9-L12

And `NonUnicodePath` is also defined for `bcf`, but `FileNotFound` is missing there. So such high-level definition could also help clean up the errors a bit:

https://github.com/rust-bio/rust-htslib/blob/61b0644b73ab40612cd23e104ff0a3fe647bf064/src/bcf/errors.rs#L40-L41

**Comment by dlaehnemann** [comment]
While this change happens, we could also think about moving the `path_as_bytes` function up to `src/utils.rs`:

https://github.com/rust-bio/rust-htslib/blob/29565f9edda5ee71e7ee6547128a76ff29b2f404/src/faidx/mod.rs#L19-L32

**Comment by landesfeind** [comment]
As discussed, I started working on this. First merge is available in [landesfeind:215_generic_errors_for_entire_crate](https://github.com/landesfeind/rust-htslib/tree/215_generic_errors_for_entire_crate)

While there are clearly some generic errors, others are also quite specific to the modules. I am wondering if there is a preference in using specialized errors with a prefix (i.e., what I did in a first merge to solve name-conflicts) like `BcfSeek` or if we should try to harmonize/generalize them (e.g., `GenomicSeek { target: String, position: u64 }` versus `FileSeek`)

Any thoughts?
